src/test/unit/StrategyBaseTVLLimitsUnit.sol

```diff
@@ -0,0 +1,114 @@
+
+    function setUp() virtual public {
+        proxyAdmin = new ProxyAdmin();
+
+        pauserRegistry = new PauserRegistry(pauser, unpauser);
+
+        strategyManager = new StrategyManagerMock();
+
+        underlyingToken = new ERC20PresetFixedSupply("Test Token", "TEST", initialSupply, initialOwner);
+
+        strategyBaseTVLLimitsImplementation = new StrategyBaseTVLLimits(strategyManager);
+
+        strategy = StrategyBaseTVLLimits(
+            address(
+                new TransparentUpgradeableProxy(
+                    address(strategyBaseTVLLimitsImplementation),
+                    address(proxyAdmin),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, underlyingToken, pauserRegistry)
+                )
+            )
+        );
+    }
+
+    function testSetTVLLimits(uint256 maxDeposits, uint256 maxPerDeposit) public {
+        cheats.startPrank(pauser);
+        strategy.setTVLLimits(maxPerDeposit, maxDeposits);
+        cheats.stopPrank();
+        (uint256 _maxPerDeposit, uint256 _maxDeposits) = strategy.getTVLLimits();
+
+        assertEq(_maxPerDeposit, maxPerDeposit);
+        assertEq(_maxDeposits, maxDeposits);
+    }
+
+    function testSetTVLLimitsFailsWhenNotCalledByPauser(uint256 maxDeposits, uint256 maxPerDeposit, address notPauser) public {
+        cheats.assume(notPauser != pauser);
+        cheats.startPrank(notPauser);
+        cheats.expectRevert(bytes("msg.sender is not permissioned as pauser"));
+        strategy.setTVLLimits(maxPerDeposit, maxDeposits);
+        cheats.stopPrank();
+    }
+
+    function testDepositMoreThanMaxPerDeposit(uint256 maxDeposits, uint256 maxPerDeposit, uint256 amount) public {
+        cheats.assume(amount > maxPerDeposit);
+        cheats.startPrank(pauser);
+        strategy.setTVLLimits(maxPerDeposit, maxDeposits);
+        cheats.stopPrank();
+
+        cheats.startPrank(address(strategyManager));
+        cheats.expectRevert(bytes("StrategyBaseTVLLimits: max per deposit exceeded"));
+        strategy.deposit(underlyingToken, amount);
+        cheats.stopPrank();
+    }
+
+    function testDepositMorthanMaxDeposits() public {
+
+        maxDeposits = 1e12;
+        maxPerDeposit = 3e11;
+        uint256 numDeposits = maxDeposits / maxPerDeposit;
+
+        underlyingToken.transfer(address(strategyManager), maxDeposits);
+        cheats.startPrank(pauser);
+        strategy.setTVLLimits(maxPerDeposit, maxDeposits);
+        cheats.stopPrank();
+
+        cheats.startPrank(address(strategyManager));
+        for (uint256 i = 0; i < numDeposits-1; i++) {
+            underlyingToken.transfer(address(strategy), maxPerDeposit);
+            strategy.deposit(underlyingToken, maxPerDeposit);
+        }
+        underlyingToken.transfer(address(strategy), maxPerDeposit);
+        cheats.expectRevert(bytes("StrategyBaseTVLLimits: max deposits exceeded"));
+        strategy.deposit(underlyingToken, maxPerDeposit);
+        cheats.stopPrank();
+    }
```

Comment:
prob we should add a test the a deposit below or equal to max per deposit works? and that a list of amounts less than max per deposit and sum to less than max deposits works?